src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -383,6 +383,46 @@ public StringSegment Trim()
         }
 
         /// <summary>
+        /// Removes all leading whitespaces.
+        /// </summary>
+        /// <returns>The trimmed <see cref="StringSegment"/>.</returns>
+        public StringSegment TrimStart()
```

Comment:
Is it a good idea to call new StringSegment() twice? I'm trying to avoid that for performance.